components/script/dom/document.rs

```diff
@@ -2152,14 +2155,16 @@ impl Document {
         // See https://github.com/servo/servo/issues/22507
         let not_ready_for_load = self.loader.borrow().is_blocked() ||
             !self.is_fully_active() ||
-            is_in_delaying_load_events_mode;
+            is_in_delaying_load_events_mode ||
+            // In case we have already aborted this document and receive a
+            // a subsequent message to load the document
+            self.loader.borrow().events_inhibited();
 
         if not_ready_for_load {
             // Step 6.
             return;
         }
 
-        assert!(!self.loader.borrow().events_inhibited());
```

Comment:
I think this assertion was wrong, mostly because `abort()` also calls `self.loader.borrow_mut().inhibit_events();`. I am not sure why `abort()` does that (it's not in the spec and the commit that introduced it is 9 years old). Still, when messages are received out-of-order and we abort and then enter this method, we should guard against that. Therefore, I think it is safe to add to the if above like with the other expressions.